includes/payments/class-edd-payment.php

```diff
@@ -0,0 +1,229 @@
+<?php
+/**
+ * Payments
+ *
+ * This class is for working with payments in EDD.
+ *
+ * @package     EDD
+ * @subpackage  Classes/Payment
+ * @copyright   Copyright (c) 2014, Pippin Williamson
+ * @license     http://opensource.org/licenses/gpl-2.0.php GNU Public License
+ * @since       2.2
+ */
+
+
+// Exit if accessed directly
+if( ! defined( 'ABSPATH' ) ) exit;
+
+
+/**
+ * EDD_Payment Class
+ *
+ * @since 2.2
+ */
+class EDD_Payment {
+
+	/**
+	 * The Payment we are working with
+	 *
+	 * @var int
+	 * @access private
+	 * @since 2.2
+	 */
+	private $payment_id;
+
+
+	/**
+	 * Setup the EDD Payments class
+	 *
+	 * @since 2.2
+	 * @param int $payment_id A given payment
+	 * @return mixed void|false
+	 */
+	public function __construct( $payment_id = false ) {
+		// If no payment ID is specified, create a new payment
+		if( ! $payment_id ) {
+			$args = array(
+				'post_title'    => '',
+				'post_status'   => 'pending',
+				'post_type'     => 'edd_payment'
+			);
+
+			$payment_id = wp_insert_post( $args );
+		}
+
+		// Store the payment ID for later use
+		if( $payment_id ) {
+			$this->payment_id = $payment_id;
+        } else {
+            $this->payment_id = 0;
+		}
+	}
+
+
+	/**
+	 * Add a download to a given payment
+	 *
+	 * @since 2.2
+	 * @param int $download_id The download to add
+	 * @param int $args Other arguments to pass to the function
+	 * @return void
+	 */
+	public function add_download( $download_id = 0, $args = array() ) {
+		$download = get_post( $download_id );
```

Comment:
Why don't we have a wrapper for the download class for something like this that returns a Download object if there, else false since the default is to make a new download if the download id doesn't exist?